### ISSUE TITLE
Cache Issues

### ISSUE THREAD

**Issue by daanieel** [spatie/laravel-permission#1250]
Hello I am having trouble configuring the cache.

Laravel: 6.2.0
Redis: 5.0.6
`laravel-permission`: 3.0

I see in [laravel-debugbar](https://github.com/barryvdh/laravel-debugbar) and database queries, but Laravel's cache works normally. What may I be doing wrong in the configuration?

**Comment by drbyte** [comment]
I'm not sure I understand your question.

What is happening that you're not expecting? What were you expecting?

**Comment by daanieel** [comment]
I thought that the result of the queries was saved in the cache, not to make several requests to the database. It's not like this ?

> I'm not sure I understand your question.
> 
> What is happening that you're not expecting? What were you expecting?

**Comment by drbyte** [comment]
Yes, the package registrar caches the permissions when retrieved from the database.
https://github.com/spatie/laravel-permission/blob/e3821559c69b2b1ea8fdd967fa89ba13a01ead78/src/PermissionRegistrar.php#L132-L135

**Comment by daanieel** [comment]
> Yes, the package registrar caches the permissions when retrieved from the database.
> https://github.com/spatie/laravel-permission/blob/e3821559c69b2b1ea8fdd967fa89ba13a01ead78/src/PermissionRegistrar.php#L132-L135

Do I have to enable this feature in any configuration? Because, it's not working here.

**Comment by drbyte** [comment]
Shouldn't need any special configuration. It uses the cache driver you've configured in your application.

**Comment by daanieel** [comment]
> Shouldn't need any special configuration. It uses the cache driver you've configured in your application.

One question does it save on cache the permission and roles of each model? Example:

`laravel_cache: user. <user-id>`

Every time the page is loaded, to get the permissions of users it makes two queries. I wanted it to allocate the result of these queries in cache to not do them.

When I search Redis for the permission keys I find nothing.

**Comment by daanieel** [comment]
**I solved the problem:**
Create an Observer for a user model, which caches the user's permission and roles.

```
public function retrieved($user)
{
    $user->roles;
    $user->permissions = $user->getAllPermissions();
    Cache::add("user.{$user->id}", $user, 7200);
}
```

So all `laravel-permission` functions do a cache query not in MySQL.
